lib/pry/pry_instance.rb

```diff
@@ -529,28 +529,29 @@ def should_print?
   # @return [String] The prompt.
   def select_prompt
     object = current_binding.eval('self')
-
     open_token = @indent.open_delimiters.any? ? @indent.open_delimiters.last :
       @indent.stack.last
 
-    c = Pry::Config.from_hash({
-                       :object         => object,
-                       :nesting_level  => binding_stack.size - 1,
-                       :open_token     => open_token,
-                       :session_line   => Pry.history.session_line_count + 1,
-                       :history_line   => Pry.history.history_line_count + 1,
-                       :expr_number    => input_array.count,
-                       :_pry_          => self,
-                       :binding_stack  => binding_stack,
-                       :input_array    => input_array,
-                       :eval_string    => @eval_string,
-                       :cont           => !@eval_string.empty?})
+    c = Pry::Config.new(nil).tap {|conf|
+      conf.merge!({
```

Comment:
Can't we keep `from_hash` and call this inside it?

Reply:
we could but better to not use `tap` at all. I will think of a new API for creating a Pry::Config non-recursively. see https://github.com/pry/pry/issues/1726